openreview/conference/invitation.py

```diff
@@ -335,11 +336,40 @@ def __init__(self, conference, reveal_authors, reveal_submission, hide_fields=[]
             }
         )
 
-class PaperWithdrawInvitation(openreview.Invitation):
 
-    def __init__(self, conference, note, reveal_authors, reveal_submission, email_pcs, hide_fields=None):
+class WithdrawSuperInvitation(openreview.Invitation):
 
+    def __init__(self, conference):
         content = invitations.withdraw.copy()
+        exp_date = tools.datetime_millis(conference.submission_stage.withdraw_submission_exp_date) if conference.submission_stage.withdraw_submission_exp_date else None
+        super(WithdrawSuperInvitation, self).__init__(
+            id=conference.get_invitation_id("Withdraw"),
+            cdate=None,
+            duedate=None,
+            expdate=exp_date,
+            readers=['everyone'],
+            writers=[conference.support_user],
+            signatures=[conference.support_user],
+            reply={
+                'forum': None,
+                'replyto': None,
+                'readers': {
+                        'values': ['everyone']
+                    },
+                'writers': {
+                    'values': [],
+                },
+                'signatures': {
+                    'values-regex': '~.*|{}'.format(conference.support_user)
+                },
+                'content': content
```

Comment:
could you set the process function here in the super invitation? then it will be easier to update from here rather than update every child invitation.

Reply:
There are some paper-specific (author_id, reviewers_id, etc.) things that are set while creating child invitations.

Reply:
Ohhh I see, would you mind to change those constants with the following:

```
 PAPER_AUTHORS_ID = f'aclweb.org/ACL/ARR/2021/September/Paper{number}/Authors'
 PAPER_REVIEWERS_ID = f'aclweb.org/ACL/ARR/2021/September/Paper{number}/Reviewers'
 PAPER_AREA_CHAIRS_ID = f'aclweb.org/ACL/ARR/2021/September/Paper{number}/Area_Chairs'
```

then you can do:

```
PAPER_AUTHORS_ID.format(number=forum_note.number)
```

for build the value of the constant you can use the method in the conference class:

```
self.get_authors_id(number='{number}')
```

Reply:
This works everywhere `forum_note.number` is required, however, `SUBMISSION_READERS = note.readers` would not work this way. Any suggestions for that?

Reply:
Could you try assigning this instead of `note.readers`?

```
SUBMISSION_READERS =  self.submission_stage.get_readers(self, '{number}')
```

If this doesn't work then we should keep the process function in the children.

Reply:
This will require access to the conference which is not there, should I add the `request_form_id` to the process function?

Reply:
No, you can setup the variable SUBMISSION_READERS here: https://github.com/openreview/openreview-py/blob/master/openreview/conference/invitation.py#L391

You will pass an array of values that may contain a group id with `{number}` so you would need to iterate all the elements of the array and replace `{number}` with `forumNote.number`.